web/src/components/storage/VolumeForm.jsx

```diff
@@ -224,6 +247,16 @@ const SizeRange = ({ errors, formData, onChange }) => {
   );
 };
 
+// constants need to be marked for translation with N_() and translated with _() later
+const SIZE_OPTION_LABELS = Object.freeze({
+  // TRANSLATORS: radio button label, fully automatically computed partition size, no user input
+  auto: N_("Auto"),
```

Comment:
Question: why `N_` instead of `_` ?

Reply:
That constant is evaluated too early, at that point the `cockpit.language` is not initialized yet and the text would not be translated.

Reply:
Ahh, thanks for the explanation.

Reply:
Hi guys!

Yesterday I found this looking more information about [N_](https://github.com/openSUSE/agama/blob/704b693ca563e48f6ead2481fa5e73224fa431c6/web/src/i18n.js#L94-L127) when reviewing [ancorgs's PR](https://github.com/openSUSE/agama/pull/1090/files#diff-0b738eb119a000a6ab33ab9e5234cc455461e03d12148e47a866140b2d3ec247R51). I know that `N_` is used to mark for translation, but I do not understand why it is needed either, here and in the @ancorgs's PR linked above.

Also, I wonder if, in case of really needed, this will change with the [new architecture](https://github.com/openSUSE/agama/discussions/1000).

Reply:
The `N_` macro/function used for translating constants, it does not depend on the architecture or programming language (you can use use it even in C or C++ code).

The problem is that 
```
Object.freeze({
  auto: _("Auto"),
   ...
```

would translate the value using the language configured at the initialization, it would not retranslate the text after changing the language later at runtime.

https://yastgithubio.readthedocs.io/en/latest/localization/#translating-constants